src/web/mjs/connectors/templates/WordPressMadara.mjs

```diff
@@ -88,23 +88,37 @@ export default class WordPressMadara extends Connector {
         // TODO: setting this parameter seems to be problematic for various website (e.g. ChibiManga, AniMangaES server will crash)
         uri.searchParams.set('style', 'list');
         let request = new Request(uri, this.requestOptions);
-        let data = await this.fetchDOM(request, this.queryPages);
+        let data;
+        if (this.queryPages instanceof Array) {
+            let data1 = await this.fetchDOM(request, 'body');
+            this.queryPages.forEach(ele => {
+                if (data1[0].querySelectorAll(ele).length > 0) {
+                    data = [...data1[0].querySelectorAll(ele)];
+                }
+            });
+        } else {
+            data = await this.fetchDOM(request, this.queryPages);
+        }
         return data.map(element => this.createConnectorURI({
             // HACK: bypass 'i0.wp.com' image CDN to ensure original images are loaded directly from host
             url: this.getAbsolutePath(element.dataset['src'] || element['srcset'] || element, request.url).replace(/\/i\d+\.wp\.com/, ''),
             referer: request.url
         }));
     }
 
-    _handleConnectorURI(payload) {
+    async _handleConnectorURI(payload) {
         /*
          * TODO: only perform requests when from download manager
          * or when from browser for preview and selected chapter matches
          */
-        this.requestOptions.headers.set('x-referer', payload.referer);
-        let promise = super._handleConnectorURI(payload.url);
-        this.requestOptions.headers.delete('x-referer');
-        return promise;
+        let request = new Request(payload.url, this.requestOptions);
+        request.headers.set('x-referer', payload.referer);
+        request.headers.set('accept', 'image/webp,image/apng,image/*,*/*');
```

Comment:
Could be removed in case this is checked for each request globally...

Reply:
this is the whole reason this PR exists 
if this isn't here the sites respond with some html rather than the image

Reply:
note: this doesn't seem to happen for everyone